<launch>
    <param name="robot_description"
    command="xacro '$(find cga_robotics_ros)/model/delta/ros.urdf.xacro'"/>

    <rosparam
        file="$(find cga_robotics_ros)/config/delta_controllers.yaml"
        command="load"
    />

    <param name="robot_description"
        command="xacro '$(find cga_kinematics)/model/delta/ros.urdf.xacro'"/>

    <rosparam file="$(find cga_kinematics)/config/loops.yaml" />

    <node
        pkg="cga_kinematics"
        type="delta_fk"
        name="delta_fk"
        output="screen">
        <remap from="joint_states_out" to="joint_states"/>
    </node>

    <node
        pkg="robot_state_publisher"
        type="robot_state_publisher"
        name="robot_state_publisher">
    </node>

    <node
        name="controller_spawner"
        pkg="controller_manager"
        type="spawner"
        respawn="false"
        args="theta_1_controller theta_2_controller theta_3_controller joint_state_controller"
    />

    <node
        pkg="cga_kinematics"
        type="delta_ik"
        name="delta_ik"
        output="screen">
    </node>

    <node
        name="rqt_gui"
        pkg="rqt_gui"
        type="rqt_gui"
        args="--perspective-file $(find cga_kinematics)/config/rqt_ik.perspective"
    />

    <node
        name="rviz" pkg="rviz" type="rviz"
        args="-d $(find cga_kinematics)/config/default.rviz"
    />
</launch>
